src/main/java/com/atlassian/bitbucket/jenkins/internal/deployments/DeploymentPosterImpl.java

```diff
@@ -67,8 +110,8 @@ public void postDeployment(String bbsServerId, String projectKey, String reposit
         taskListener.getLogger().println(format("Sending notification of %s to %s on commit %s",
                 deployment.getState().name(), server.getServerName(), revisionSha));
         try {
-            clientFactory.getProjectClient(projectKey)
-                    .getRepositoryClient(repositorySlug)
+            clientFactory.getProjectClient(bitbucketSCMRepo.getProjectKey())
```

Comment:
did you try this with a personal repository as well? (it should be fine, but we should probably try it if you haven't already)